pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -331,12 +345,59 @@ private static void queryRewrite(PinotQuery pinotQuery) {
       pinotQuery.setFilterExpression(updatedFilterExpression);
     }
 
+    // Rewrite GroupBy to Distinct
+    rewriteNonAggregationGroupByToDistinct(pinotQuery);
+
     // Update alias
     Map<Identifier, Expression> aliasMap = extractAlias(pinotQuery.getSelectList());
     applyAlias(aliasMap, pinotQuery);
     validate(aliasMap, pinotQuery);
   }
 
+  private static void rewriteNonAggregationGroupByToDistinct(PinotQuery pinotQuery) {
```

Comment:
Would be good to add doc with sample query translations (as already stated in the PR description).

Reply:
will do